java/RSPEC-1764.java

```diff
@@ -0,0 +1,14 @@
+class IdentExprBinary {
+    void test() {
+        int a = 1;
+        int b = 2;
+        if (a == a) {}
+        if (a == b && a == b) {}
+        if (a == b && b == a) {}
+        this.equals(this);
+        a = b + b;
```

Comment:
RSPEC-2757: Don't repeat Unary operators
<!-- lookout footnote separator -->
_Comment made by the analyzer sonarcheck_analyzer._